feat: adds experimental features support with dark mode toggle

av3n93rz opened the pull request:
Introduces state management for experimental features and a dialog accessible via keyboard shortcut, allowing users to enable or disable features such as dark mode. Refactors global providers to improve theme and state handling, simplifying layout logic and enhancing extensibility for future experiments.

## Checklist

- [ ] Follow the [contributor guide](./CONTRIBUTING.md)
- [ ] End-to-end tests pass
- [ ] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [ ] Update `./docs`
- [ ] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [ ] Linked issues #eg101 

The experimental features dialog is accessible by pressing `ctrl/super + e` when the experimental features are turned on.
To enable experimental features set `experimental-features-enabled: true` in localstorage.

https://github.com/user-attachments/assets/ad52e131-283e-4bc3-8620-9913fc5f6b7f

<img width="1917" height="929" alt="Screenshot 2025-10-10 at 16 53 01" src="https://github.com/user-attachments/assets/bfb46621-4dbf-4143-931f-14839ae50bfe" />
<img width="1917" height="929" alt="Screenshot 2025-10-10 at 16 52 51" src="https://github.com/user-attachments/assets/c4f6637f-3f33-4685-b241-57102c34d9a1" />

av3n93rz committed:
feat: adds experimental features support with dark mode toggle

Introduces state management for experimental features and a dialog accessible via keyboard shortcut, allowing users to enable or disable features such as dark mode. Refactors global providers to improve theme and state handling, simplifying layout logic and enhancing extensibility for future experiments.

dwmkerr reviewed (commented):
Looks good to me - should we have a brief note in the ask dashboard docs?

av3n93rz committed:
docs: adds experimental features section to user guide

Introduces a new entry for experimental features in the user documentation to clarify feature maturity and guide users on preview functionalities.

av3n93rz commented:
> Looks good to me - should we have a brief note in the ask dashboard docs?

Added a short section to the user guide
